main/keychain/src/main/java/org/cryptomator/keychain/LinuxSystemKeychainAccess.java

```diff
@@ -13,28 +17,66 @@
 @Singleton
 public class LinuxSystemKeychainAccess implements KeychainAccessStrategy {
 
-	// the actual implementation is hidden in this delegate objects which are loaded via reflection,
-	// as it depends on libraries that aren't necessarily available:
+	// the actual implementation is hidden in this delegate object,
+	// as on Linux the are two possible password backends available:
 	private final Optional<KeychainAccessStrategy> delegate;
+	private final Settings settings;
+	private static EnumSet<KeychainBackend> availableKeychainBackends = EnumSet.noneOf(KeychainBackend.class);
+	private static KeychainBackend backendActivated = null;
+	private static boolean isGnomeKeyringAvailable;
+	private static boolean isKdeWalletAvailable;
 
 	@Inject
-	public LinuxSystemKeychainAccess() {
+	public LinuxSystemKeychainAccess(Settings settings) {
+		this.settings = settings;
 		this.delegate = constructKeychainAccess();
 	}
 
-	private static Optional<KeychainAccessStrategy> constructKeychainAccess() {
-		try { // is gnome-keyring or kwallet installed?
-			Class<?> clazz = Class.forName("org.cryptomator.keychain.LinuxSecretServiceKeychainAccessImpl");
-			KeychainAccessStrategy instance = (KeychainAccessStrategy) clazz.getDeclaredConstructor().newInstance();
-			if (instance.isSupported()) return Optional.of(instance);
-			clazz = Class.forName("org.cryptomator.keychain.LinuxKDEWalletKeychainAccessImpl");
-			instance = (KeychainAccessStrategy) clazz.getDeclaredConstructor().newInstance();
-			return Optional.of(instance);
+	private Optional<KeychainAccessStrategy> constructKeychainAccess() {
+		try { // find out which backends are available
+			KeychainAccessStrategy gnomeKeyring = new LinuxSecretServiceKeychainAccessImpl();
```

Comment:
The reason why we used reflection was that `LinuxSecretServiceKeychainAccessImpl` imports classes not available on non-linux platforms.

Reply:
Thinking about this, it might be actually worth refactoring this to use the `ServiceLoader` API... 🤔

Reply:
> The reason why we used reflection was that `LinuxSecretServiceKeychainAccessImpl` imports classes not available on non-linux platforms.

That's right. I was under the impression, that  `LinuxSystemKeychainAccess` is not called on non-Linux systems, but that's wrong.

Reply:
As soon as LinuxSystemKeychainAccess is _imported_, you should consider it being used. The classloader will attempt to recursively satisfy all imports. Therefore this needs to be reverted to reflection for now.